include/veriblock/blockchain/commands/addvtb.hpp

```diff
@@ -25,11 +26,28 @@ struct AddVTB : public Command {
     // add commands to VBK containing block
     auto containing = vtb_.containingBlock.getHash();
     auto& vbk = tree_->vbk();
+    auto* index = vbk.getBlockIndex(containing);
+    if (!index) {
+      return state.Invalid(
+          block_t::name() + "-bad-containing",
+          "Can not find VTB containing block: " + containing.toHex());
+    }
+
+    // duplicates not allowed
+    if (std::find(index->payloadIds.begin(),
+                  index->payloadIds.end(),
+                  vtb_.getId()) != index->payloadIds.end()) {
+      return state.Invalid(
+          block_t::name() + "-duplicate-payloads",
+          fmt::sprintf("Containing block=%s already contains payload %s.",
+                       index->toPrettyString(),
+                       vtb_.getId().toHex()));
+    }
```

Comment:
addPayloads should handle duplicates by itself, why is this needed?

Reply:
Not sure it handles them properly. In scenario5 they were added without any error message but failed when tried to erase endorsedBy multiple times.